### ISSUE TITLE
Support HTML emails (for newsletter & registration e.g.)

### ISSUE THREAD

**Issue by Total-Reality** [contao/contao#3906]
### Description

The default text emails are very ugly.
It is currently not possible to send HTML emails in the contao core modules in an easy way. It is possible to do this, but I need to clone the whole ModuleSubscribe.php and also parts of the OptIn Logic. This is extreme for such a simple thing... it is too complicated.

So can you just add three new lines her please?

https://github.com/contao/core-bundle/blob/4d6540349a8858eebc415b6ee6fb2c34d55704c3/src/OptIn/OptInToken.php#L137
`
if($html) {
$email->html = $html;
}
`

Additionally change this line:
https://github.com/contao/core-bundle/blob/4d6540349a8858eebc415b6ee6fb2c34d55704c3/src/OptIn/OptInToken.php#L114
(adding a third string value in the function)

`public function send(string $subject = null, string $text = null, string $html): void`

**Comment by fritzmg** [comment]
> but I need to clone the whole ModuleSubscribe.php and also parts of the OptIn Logic.

Or use the [Notification Center](https://github.com/terminal42/contao-notification_center).

**Comment by Total-Reality** [comment]
No sorry, I don't like this extension. It's too complicate.
My wish is a simple "feature" which should be part of the core. Why should I install an extension which installs so many files und reduces my performance?

**Comment by m-vo** [comment]
> Why should I install an extension which installs so many files und reduces my performance?

Could you please stop with these bold and false claims? Your initial feature request itself is valid and having functionality, like the notification center provides, as part of the core, is a wish many have. I'd be :+1: if anyone finds the time and energy to migrate it. But sentences like this really do not want me to look any further.

**Comment by fritzmg** [comment]
Generally I agree though that the emails sent by the core lack the ability to be customised. For instance only `contao/newsletter-bundle` actually uses the `mail_default` template. Everything else sends text emails with the bare minimum.

**Comment by Total-Reality** [comment]
> > Why should I install an extension which installs so many files und reduces my performance?
> 
> Could you please stop with these bold and false claims? Your initial feature request itself is valid and having functionality, like the notification center provides, as part of the core, is a wish many have. I'd be 👍 if anyone finds the time and energy to migrate it. But sentences like this really do not want me to look any further.

It's not a good argument, that sth. should not be implemented, because there exists an extension. Contao is much better than any other CMS, because very nice features are implemented in the core system already.

I'm sorry, but these are not false claims. I made it clear that I **personally** don't like this extension. But you are also quite subjective. There are often problems with third-party extensions, it is better if some things are already included in the core.
Core functions are much more sophisticated and can cover an extremely large number of use cases.

I don't necessarily think that all functions should be integrated from NC, because the settings are sometimes very confusing and nested. But some functions are certainly useful.

In the past we had a lot of problems with some extensions such as Notification Center and "Haste" at. As a result, entire pages were shot down completely.
It should also not be denied that every additional extension results in minimal performance losses. It's logical... more code + more modules + more DCA et cetera = more bytes that are transmitted. Yeah okay... it may be very minimal, but it is like that. I already have more than 25 extensions in some installations, I really don't want to install anything else that can be covered by just a few lines of code in the core.

> Generally I agree though that the emails sent by the core lack the ability to be customised. For instance only `contao/newsletter-bundle` actually uses the `mail_default` template. Everything else sends text emails with the bare minimum.

That's right, good point. Shouldn't this logic be carried further? Wouldn't it be logical if the mail_default template could be used for all other areas also?

**Comment by ausi** [comment]
> It should also not be denied that every additional extension results in minimal performance losses.

I was not able to reproduce that. How did you measure these performance losses? And what are the exact results?

> It's logical... more code + more modules + more DCA et cetera = more bytes that are transmitted.

If that would be the case, adding your few lines of code would result in performance losses for every user of Contao, right?

**Comment by Total-Reality** [comment]
To take a simple example: Are you suggesting me that a PHP file or another script with 10,000 lines of code doesn't load faster and isn't processed faster than a script file with 1,000,000 lines of code?

I know it doesn't have these dimensions, but small cattle also make crap.

**Comment by fritzmg** [comment]
> To take a simple example: Are you suggesting me that a PHP file or another script with 10,000 lines of code doesn't load faster and isn't processed faster than a script file with 1,000,000 lines of code?

This discussion is unrelated to this issue. Please discuss this in a community of your choice, not here.

**Comment by Total-Reality** [comment]
Yeah, you are right. Thank you, I also don't want to discuss this here. Your mates m-vo and ausi asked me some provocative questions. Sorry for my answers. OMG.

Back to topic: I can create a pull request perhaps.

**Comment by ameotoko** [comment]
> Why should I install an extension which installs so many files und reduces my performance?

I, for one, can't wait when NC will finally be in core (or at least one of `contao/` bundles) 😁
